src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/PEMethodSymbol.vb

```diff
@@ -641,7 +641,26 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
             If Not _packedFlags.IsCustomAttributesPopulated Then
                 Dim attributeData As ImmutableArray(Of VisualBasicAttributeData) = Nothing
                 Dim containingPEModuleSymbol = DirectCast(ContainingModule(), PEModuleSymbol)
-                containingPEModuleSymbol.LoadCustomAttributes(Me.Handle, attributeData)
+                Dim checkForRequiredMembers = MethodKind = MethodKind.Constructor AndAlso
+                                              HasSetsRequiredMembers = False AndAlso
+                                              (Me.ContainingType.HasAnyDeclaredRequiredMembers OrElse Not Me.ContainingType.AllRequiredMembers.IsEmpty)
+
+                If checkForRequiredMembers Then
+                    Dim compilerFeatureRequiredDiagnostic As DiagnosticInfo = Nothing
+                    DeriveCompilerFeatureRequiredUseSiteInfo(compilerFeatureRequiredDiagnostic)
+
+                    Dim discard1 As CustomAttributeHandle = Nothing
+                    Dim discard2 As CustomAttributeHandle = Nothing
+                    attributeData = containingPEModuleSymbol.GetCustomAttributesForToken(
+                        Me.Handle,
+                        filteredOutAttribute1:=discard1,
+                        filterOut1:=If(compilerFeatureRequiredDiagnostic Is Nothing, AttributeDescription.CompilerFeatureRequiredAttribute, Nothing),
+                        filteredOutAttribute2:=discard2,
+                        filterOut2:=If(ObsoleteAttributeData Is Nothing, AttributeDescription.ObsoleteAttribute, Nothing))
```

Comment:
>If(ObsoleteAttributeData Is Nothing, AttributeDescription.ObsoleteAttribute, Nothing) [](http://example.com/codeflow?start=36&length=85)

Does this filter out explicit `[Obsolete]` attributes as well (for a constructor on a type with a `required` member that has been deprecated)?